app.py

```
@@ -250,13 +253,13 @@ def _get_api_key_dynamic(api_key: str = Depends(api_key_header)):
         # Preserve HTTPException semantics (e.g., 403 for auth), convert other errors to 500
         if isinstance(exc, HTTPException):
             raise
-        raise HTTPException(status_code=500, detail=f"auth dependency error: {exc}")
+        logger.error("Auth dependency failure occurred")
```

Ruff (0.13.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)